test/tests/AelinFeeDistributor.t.sol

```diff
@@ -0,0 +1,298 @@
+        emit Transfer(address(feeDistributor), user1, claimableToken4);
+        vm.expectEmit(false, false, false, true);
+        emit Claimed(0, user1, share);
+        feeDistributor.claim(0, user1, share, merkleProof);
+
+        assertEq(MockERC20(token1).balanceOf(user1), claimableToken1);
+        assertEq(MockERC20(token2).balanceOf(user1), claimableToken2);
+        assertEq(MockERC20(token3).balanceOf(user1), claimableToken3);
+        assertEq(MockERC20(token4).balanceOf(user1), claimableToken4);
+
+        vm.expectRevert("MerkleDistributor: Drop already claimed.");
+        feeDistributor.claim(0, user1, share, merkleProof);
+        vm.stopPrank();
+    }
+
+    function test_Claim_RevertWhen_InvalidProof() public {
+        vm.startPrank(user1);
+        uint256 share = 500000000000000000;
+
+        bytes32[] memory merkleProof = new bytes32[](2);
+        merkleProof[0] = 0x0a7e356738486e70683cadf436a6f14789f356637327b190e17ab338af7f0910;
+        merkleProof[1] = 0xac1fe0856a9ea8b0b37641df7cdd623f456573a64be32e4b28bb7bdf48024e48;
+
+        vm.expectRevert("MerkleDistributor: Invalid proof.");
+        feeDistributor.claim(1, user1, share, merkleProof);
+
+        vm.expectRevert("MerkleDistributor: Invalid proof.");
+        feeDistributor.claim(0, user2, share, merkleProof);
+
+        vm.expectRevert("MerkleDistributor: Invalid proof.");
+        feeDistributor.claim(0, user1, 1, merkleProof);
+
+        vm.expectRevert("MerkleDistributor: Invalid proof.");
+        merkleProof[1] = 0xac1fe0856a9ea8b0b37641df7cdd623f456573a64be32e4b28bb7bdf48024e43;
+        feeDistributor.claim(0, user1, 1, merkleProof);
+
+        vm.stopPrank();
+    }
+
+    function test_Claim() public {
+        // user1 claims
+        vm.startPrank(user1);
+        uint256 share = 500000000000000000;
+        uint256 claimableToken1 = (share * feeDistributor.TOKEN1_AMOUNT()) / 1e18;
+        uint256 claimableToken2 = (share * feeDistributor.TOKEN2_AMOUNT()) / 1e18;
+        uint256 claimableToken3 = (share * feeDistributor.TOKEN3_AMOUNT()) / 1e18;
+        uint256 claimableToken4 = (share * feeDistributor.TOKEN4_AMOUNT()) / 1e18;
+
+        bytes32[] memory merkleProof = new bytes32[](2);
+        merkleProof[0] = 0x0a7e356738486e70683cadf436a6f14789f356637327b190e17ab338af7f0910;
+        merkleProof[1] = 0xac1fe0856a9ea8b0b37641df7cdd623f456573a64be32e4b28bb7bdf48024e48;
+        vm.expectEmit(true, false, false, true);
+        emit Transfer(address(feeDistributor), user1, claimableToken1);
+        vm.expectEmit(true, false, false, true);
+        emit Transfer(address(feeDistributor), user1, claimableToken2);
+        vm.expectEmit(true, false, false, true);
+        emit Transfer(address(feeDistributor), user1, claimableToken3);
+        vm.expectEmit(true, false, false, true);
+        emit Transfer(address(feeDistributor), user1, claimableToken4);
+        vm.expectEmit(false, false, false, true);
```

Comment:
Not sure why you're not checking all topics by doing:  `vm.expectEmit(false, false, false, true);` and not
`vm.expectEmit(true, true, true, true);`